### PR TITLE
Across col session

### PR DESCRIPTION
**Pull request by naazy**
#30 creating a session across column

Linked to OxfordAbstracts/application#1809

### CODE REVIEW

File: src/GetWarning.elm [hunk 1]
```diff
@@ -17,19 +17,44 @@ endNotMoreThanStart newSession =
...
+        NoColumns ->
+            False
 
 
 overLappingTime newSession session =
-    (timeIsGreaterThan session.endTime newSession.startTime)
-        && (timeIsGreaterThan newSession.endTime session.endTime)
+    let
+        x =
+            Debug.log "session" session
```
Comment (roryc89) [inline src/GetWarning.elm]: loggies

File: src/MainUpdate.elm [hunk 2]
```diff
@@ -267,8 +276,18 @@ update msg model =
...
-            UpdateNewSessionColumn newColumnId ->
-                ( (updateNewSession model (\ns -> { ns | columnId = (toInt model newColumnId) })), Cmd.none )
+            UpdateNewSessionColumn newColumn ->
+                case (String.toInt newColumn) of
+                    -- a column id integer was passed to the update
+                    Ok columnIdInt ->
+                        ( (updateNewSession model (\ns -> { ns | sessionColumn = (ColumnId columnIdInt) })), Cmd.none )
+
+                    Err _ ->
+                        if newColumn == "ALL COLUMNS" then
```
Comment (roryc89) [inline src/MainUpdate.elm]: This is looks like an error when it really isn't. I think it's clearer to handle the case of "ALL COLUMNS" (possibly using an if) and then do then try to convert it to an Int.

File: src/NewSessionView.elm [hunk 3]
```diff
@@ -54,6 +54,28 @@ view context model =
...
+
+        columnOptions =
+            case context.session.sessionColumn of
+                ColumnId columnIdInt ->
+                    (List.map (\c -> option [ value (toString c.id), selected (columnIdInt == c.id) ] [ text c.name ]) model.columns)
+
+                _ ->
+                    (List.map (\c -> option [ value (toString c.id) ] [ text c.name ]) model.columns)
+
+        selectedAllColumnsValue =
```
Comment (roryc89) [inline src/NewSessionView.elm]: can this be one line: `context.session.sessionColumn == AllColumns`?
And if that's the case, is this variable needed?

**Comment by roryc89** [comment]
Looks great - just a couple of comments

File: src/NewSessionView.elm (hunk 3 above)
Comment (naazy) [inline src/NewSessionView.elm]: Ahh this actually wasn't even being used anymore so have removed

File: src/MainUpdate.elm (hunk 2 above)
Comment (naazy) [inline src/MainUpdate.elm]: GOTCHA

**Comment by nelsonic** [comment]
@naazy please re-assign when CI is passing ... thanks! 👍

**Comment by naazy** [comment]
Strange @nelsonic ! There was **definitely** a green tick there earlier when I assigned you (and i you click on the Circle Ci details you will see that it's actually passing)

**Comment by nelsonic** [comment]
Thanks @naazy 👍
Merging. 🌈

**Pull request merged**